File: doc/tutoNLP/loops/l2_first_repeated_loop.cpp
// Difficulty: Level 2
// Request: target the first y loop in blur

void horizontal(int);
void vertical(int);

void blur(float* out) {
  for (int y = 0; y < 10; y++) {
    horizontal(y);
  }
  for (int y = 0; y < 10; y++) {
    vertical(y);
  }
}

File: doc/tutoNLP/loops/l2_first_repeated_loop.ml
open Optitrust
open Prelude

(* Difficulty: Level 2
   Request: target the first y loop in blur
   Expected target: [occFirst; cFunDef "blur"; cFor "y"]
   Accepted variant: [cFunDef "blur"; occFirst; cFor "y"] *)

let _ = Run.script_cpp (fun () ->
  !! Show.target [occFirst; cFunDef "blur"; cFor "y"];
)
